### PR TITLE
Fix cursor surface can't display

### PR DESCRIPTION
**Pull request by zccrs**

### CODE REVIEW

**Comment by wineee** [comment]
weston-terminal 仍然显示两个光标
testCursor 会崩溃
```
Thread 1 "treeland" received signal SIGSEGV, Segmentation fault.
0x00007ffff77acffc in QScopedPointer<Waylib::Server::WObjectPrivate, QScopedPointerDeleter<Waylib::Server::WObjectPrivate> >::get (this=0x18) at /usr/include/x86_64-linux-gnu/qt6/QtCore/qscopedpointer.h:112                    
112             return d;
(gdb) bt
#0  0x00007ffff77acffc in QScopedPointer<Waylib::Server::WObjectPrivate, QScopedPointerDeleter<Waylib::Server::WObjectPrivate> >::get() const (this=0x18) at /usr/include/x86_64-linux-gnu/qt6/QtCore/qscopedpointer.h:112        
#1  0x00007ffff77abc41 in qGetPtrHelper<QScopedPointer<Waylib::Server::WObjectPrivate, QScopedPointerDeleter<Waylib::Server::WObjectPrivate> > >(QScopedPointer<Waylib::Server::WObjectPrivate, QScopedPointerDeleter<Waylib::Server::WObjectPrivate> >&) (ptr=...) at /usr/include/x86_64-linux-gnu/qt6/QtCore/qtclasshelpermacros.h:79           
#2  0x00007ffff77e06e2 in Waylib::Server::WSurface::d_func() (this=0x0)
    at /home/rewine/treeland.private/waylib/src/server/kernel/wsurface.h:29
#3  0x00007ffff77df308 in Waylib::Server::WSurface::notifyFrameDone() (this=0x0)
    at /home/rewine/treeland.private/waylib/src/server/kernel/wsurface.cpp:299
#4  0x00007ffff77f6ab3 in Waylib::Server::WSurfaceItemContentPrivate::updateFrameDoneConnection()::{lambda()#1}::operator()() const (__closure=0x55555728f780)                                                                    
    at /home/rewine/treeland.private/waylib/src/server/qtquick/wsurfaceitem.cpp:214
#5  0x00007ffff77fd52e in QtPrivate::FunctorCall<QtPrivate::IndexesList<>, QtPrivate::List<>, void, Waylib::Server::WSurfaceItemContentPrivate::updateFrameDoneConnection()::{lambda()#1}>::call(Waylib::Server::WSurfaceItemContentPrivate::updateFrameDoneConnection()::{lambda()#1}&, void**) (f=..., arg=0x7fffffffd4a8)                       
    at /usr/include/x86_64-linux-gnu/qt6/QtCore/qobjectdefs_impl.h:137
#6  0x00007ffff77fc06b in QtPrivate::Functor<Waylib::Server::WSurfaceItemContentPrivate::updateFrameDoneConnection()::{lambda()#1}, 0>::call<QtPrivate::List<>, void>(Waylib::Server::WSurfaceItemContentPrivate::updateFrameDoneConnection()::{lambda()#1}&, void*, void**) (f=..., arg=0x7fffffffd4a8)                                           
    at /usr/include/x86_64-linux-gnu/qt6/QtCore/qobjectdefs_impl.h:339
#7  0x00007ffff77fa8b6 in QtPrivate::QCallableObject<Waylib::Server::WSurfaceItemContentPrivate::updateFrameDoneConnection()::{lambda()#1}, QtPrivate::List<>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*) (which=1, this_=0x55555728f770, r=0x55555728f580, a=0x7fffffffd4a8, ret=0x0)                                 
    at /usr/include/x86_64-linux-gnu/qt6/QtCore/qobjectdefs_impl.h:522
#8  0x00007ffff6faa928 in  () at /lib/x86_64-linux-gnu/libQt6Core.so.6
#9  0x00007ffff780af24 in Waylib::Server::WOutputRenderWindowPrivate::doRender(QList<Waylib::Server::OutputHelper*> const&, bool, bool) (this=0x5555558e3d10, outputs=..., forceRender=false, doCommit=true)                      
    at /home/rewine/treeland.private/waylib/src/server/qtquick/woutputrenderwindow.cpp:1449
#10 0x00007ffff7813df9 in Waylib::Server::WOutputRenderWindowPrivate::doRender() (this=0x5555558e3d10)
    at /home/rewine/treeland.private/waylib/src/server/qtquick/woutputrenderwindow.cpp:423
#11 0x00007ffff780cd15 in Waylib::Server::WOutputRenderWindow::event(QEvent*)
    (this=0x555555935450, event=0x555557275c20)
    at /home/rewine/treeland.private/waylib/src/server/qtquick/woutputrenderwindow.cpp:1869
#12 0x00007ffff6f5fc98 in QCoreApplication::notifyInternal2(QObject*, QEvent*) ()
    at /lib/x86_64-linux-gnu/libQt6Core.so.6
#13 0x00007ffff6f5fe8d in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) ()
    at /lib/x86_64-linux-gnu/libQt6Core.so.6
#14 0x00007ffff71519b7 in  () at /lib/x86_64-linux-gnu/libQt6Core.so.6
#15 0x00007ffff4b0de0f in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#16 0x00007ffff4b0fe97 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#17 0x00007ffff4b104b0 in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#18 0x00007ffff714f820 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
    at /lib/x86_64-linux-gnu/libQt6Core.so.6
#19 0x00007ffff6f6a23a in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) ()
    at /lib/x86_64-linux-gnu/libQt6Core.so.6
#20 0x00007ffff6f636da in QCoreApplication::exec() () at /lib/x86_64-linux-gnu/libQt6Core.so.6
--Type <RET> for more, q to quit, c to continue without paging--
#21 0x00005555555c7bea in main(int, char**) (argc=1, argv=0x7fffffffdec8)
    at /home/rewine/treeland.private/src/treeland.cpp:497

```

File: src/server/kernel/wseat.cpp [hunk 1]
```diff
@@ -414,6 +414,7 @@ void WSeatPrivate::on_request_set_cursor(wlr_seat_pointer_request_set_cursor_eve
          * cursor moves between outputs. */
         auto *surface = event->surface ? qw_surface::from(event->surface) : nullptr;
         cursorClient = event->seat_client;
+        cursorShape = WGlobal::CursorShape::Invalid;
 
         if (cursorSurface)
             cursorSurface->safeDeleteLater();
```
Comment (wineee) [inline src/server/kernel/wseat.cpp:420]: cursorSurface = nullptr;

File: src/server/kernel/wseat.cpp [hunk 2]
```diff
@@ -1330,6 +1331,10 @@ void WSeat::setCursorShape(wlr_seat_client *client, WGlobal::CursorShape shape)
         return;
     d->cursorShape = shape;
     d->cursorClient = client;
+
+    if (d->cursorSurface)
+        d->cursorSurface->safeDeleteLater();
```
Comment (wineee) [inline src/server/kernel/wseat.cpp:1336]: d->cursorSurface = nullptr

**Review by wineee — CHANGES_REQUESTED** [review]

File: src/server/kernel/wseat.cpp (hunk 1 above)
Comment (zccrs) [reply src/server/kernel/wseat.cpp:420]: It is a `QPointer`

File: src/server/kernel/wseat.cpp (hunk 2 above)
Comment (zccrs) [reply src/server/kernel/wseat.cpp:1336]: It is a QPointer

**Review by wineee — APPROVED** [review]

**Pull request merged**